app/models/points/participation.rb

```diff
@@ -0,0 +1,40 @@
+module Points
+  class Participation
+    def calculate(game, row)
+      @game = game
+      @row = row
+
+      points_for_participation.tap do |points|
+        return 0 if current >= max || week_current >= week_max
+        return (max - current) if (current + points) >= max
+        return (week_max - week_current) if (week_current + points) >= week_max
+      end
+    end
```

Comment:
Huh, `#tap` is cute. Not seen that before outside of JavaScript and it's not baked into the language like this is. (You could write it yourself easily enough or use a library such as LoDash.)

:+1: for showing me something.